saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/Saml2AuthenticationTokenConverter.java

```diff
@@ -62,6 +66,23 @@ public Saml2AuthenticationTokenConverter(
 		this.relyingPartyRegistrationResolver = relyingPartyRegistrationResolver;
 	}
 
+	/**
+	 * Constructs a {@link Saml2AuthenticationTokenConverter} with the provided parameters
+	 * @param relyingPartyRegistrationResolver the strategy for resolving
+	 * {@link RelyingPartyRegistration}s
+	 * @param authenticationRequestRepository the repository to load the
+	 * {@link AbstractSaml2AuthenticationRequest} from
+	 * @since 5.6
+	 */
+	public Saml2AuthenticationTokenConverter(
+			Converter<HttpServletRequest, RelyingPartyRegistration> relyingPartyRegistrationResolver,
+			Saml2AuthenticationRequestRepository<AbstractSaml2AuthenticationRequest> authenticationRequestRepository) {
```

Comment:
Usually, it's preferred to either have a setter or a constructor parameter, but not both. In this case, since `Saml2AuthenticationRequestRepository` is optional, I'd recommend removing the constructor parameter.